### PR TITLE
New fixes to HF FG bit

### PR DESCRIPTION
**Pull request by mnorthup6690**
Also merging with 76X #12129.

### CODE REVIEW

**Comment by mnorthup6690** [comment]
This branch includes changes to the HF FG bit definition that fix problems with the previous pull requests,
#11842 and #11687.

**Comment by velicanu** [comment]
This needs to be merged for HI data taking. We have validated that it matches data, see below.

https://twiki.cern.ch/twiki/pub/Main/MBEmulation2015/MinBiasEmulation_Comparison_2015-11-05.pdf

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

File: SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc [hunk 1]
```diff
@@ -43,7 +43,362 @@ HcalTriggerPrimitiveAlgo::HcalTriggerPrimitiveAlgo( bool pf, const std::vector<d
...
+	int GCTphi = 0;
+
+        if (phi == 1 || phi == 71)
+          {       
+                  GCTphi = 0;
+          }
+          else if (phi == 3 || phi == 5)
+            {     
+                  GCTphi = 1;
+            }
```
Comment (davidlange6) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: I think this could better be

if (phi==71) GCTphi=0;
else GCTphi=(phi+1)/4

**Comment by davidlange6** [comment]
@mnorthup6690 - and there are several other cases where the if ( phi... ) statements can be made much simpler. Could you have a look. Thanks!

**Comment by abdoulline** [comment]
Not looking at the exact relations between the values in these "else-if", I've suggested "lookup" map in a sibling PR
https://github.com/cms-sw/cmssw/pull/12129

Now I see David has spotted there are very simple functions which can possibly replace all these "trees"...

**Comment by mulhearn** [comment]
@davidlange6 @abdoulline  I propose we leave this is implemented here in 75X... it has been tested after all, and is urgently needed... and pursue the better implemented version in the 80X branch.

**Comment by abdoulline** [comment]
Improving 80X-only version sounds reasonable

**Comment by davidlange6** [comment]
@mulhearn -is that to say that testing the change by looking at dqm histograms is insufficient for knowing if the relatively straightfoward changes are ok or not? [these promised future changes in CMS tend not to happen on average:)]

**Comment by mulhearn** [comment]
@davidlange6:  Indeed, I was suggesting we don't merge the request as in the 80X branch at all, only for the 75X.  We only accept cleaned up version in 80X.

**Comment by mulhearn** [comment]
It seems 80X version is already updated...

**Comment by mulhearn** [comment]
@davidlange6  You can have a look at discussion here #12287.  The cleaned up version is available for 80X already, but we prefer to keep this version in 75X (urgently needed) as it has been more thoroughly tested.  In this case, I think we can have our cake and eat it too.

**Comment by mulhearn** [comment]
Thank you!

**Pull request merged**